src/config.py

```diff
@@ -0,0 +1,217 @@
+"""Config for solver accounting."""
+
+import os
+from dataclasses import dataclass
+from fractions import Fraction
+from pathlib import Path
+
+from eth_typing.evm import ChecksumAddress
+from dotenv import load_dotenv
+from dune_client.types import Address
+from gnosis.eth.ethereum_network import EthereumNetwork
+from web3 import Web3
+
+load_dotenv()
+
+
+@dataclass(frozen=True)
+class RewardConfig:
+    """Configuration for reward mechanism."""
+
+    reward_token_address: Address
+    cow_bonding_pool: Address
+    batch_reward_cap_upper: int
+    batch_reward_cap_lower: int
+    quote_reward_cow: int
+    quote_reward_cap_native: int
+    service_fee_factor: Fraction
+
+
+@dataclass(frozen=True)
+class ProtocolFeeConfig:
+    """Configuration for protocol and parner fees."""
+
+    protocol_fee_safe: Address
+    partner_fee_cut: float
+    partner_fee_reduced_cut: float
+    reduced_cut_address: str
+
+
+@dataclass(frozen=True)
+class BufferAccountingConfig:
+    """Configuration for buffer accounting."""
+
+    include_slippage: bool
+
+
+@dataclass(frozen=True)
+class OrderbookConfig:
+    """Configuration for orderbook fetcher"""
+
+    prod_db_url: str
+    barn_db_url: str
+
+
+@dataclass(frozen=True)
+class DuneConfig:
+    """Configuration for DuneFetcher."""
+
+    dune_api_key: str
+    dune_blockchain: str
+
+
+@dataclass(frozen=True)
+class NodeConfig:
+    """Configuration for web3 node."""
+
+    node_url: str
+
+
+@dataclass(frozen=True)
+class PaymentConfig:
+    """Configuration of payment."""
+
+    # pylint: disable=too-many-instance-attributes
+
+    network: EthereumNetwork
+    cow_token_address: Address
+    payment_safe_address: ChecksumAddress
+    signing_key: str | None
+    safe_queue_url: str
+    csv_airdrop_url: str
+    verification_docs_url: str
+    weth_address: ChecksumAddress
+
+
+@dataclass(frozen=True)
+class IOConfig:
+    """Configuration of input and output."""
+
+    log_config_file: Path
+    project_root_dir: Path
+    query_dir: Path
+    csv_output_dir: Path
+    dashboard_dir: Path
+    slack_channel: str | None
+    slack_token: str | None
+
+
+@dataclass(frozen=True)
+class AccountingConfig:
+    """Full configuration for solver accounting."""
+
+    # pylint: disable=too-many-instance-attributes
+
+    payment_config: PaymentConfig
+    orderbook_config: OrderbookConfig
+    dune_config: DuneConfig
+    node_config: NodeConfig
+    reward_config: RewardConfig
+    protocol_fee_config: ProtocolFeeConfig
+    buffer_accounting_config: BufferAccountingConfig
+    io_config: IOConfig
+
+
+def get_accounting_config(network: str) -> AccountingConfig:
```

Comment:
I think it would be easier to maintain if we move this to a classmethod, that way we instantiate the class rather than call a function that returns an object of that class. This would be best if we would want to instantiate the config in different ways (eg. config.from_network() or config.from_env()). 

If there's only ever one way, we could define a post_init that takes a string (or ENUM) to instantiate the config object. [Here is an example of how I set up the config in dune-sync with a post-init method.](https://github.com/cowprotocol/dune-sync/blob/54e22cf2c518820c66824ded0822c1fbddd765ff/src/deploy_prefect/models.py#L56) In this case I think it's better to stick to a classmethod though because the intitialization is a bit more complex.

Reply:
I moved the logic into a static class method `from_network`.